tools/tests/progress-bar/test.js

```diff
@@ -0,0 +1,137 @@
+  }
+
+  document.body.appendChild(el);
+
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve();
+    }, 1000);
+  });
+}
+
+const targetValue = '75';
+const targetValue2 = '25';
+
+function createProgressBarAsyncSpec(callback, time = 1000) {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      callback();
+      resolve();
+    }, time);
+  });
+}
+
+describe('Progress Bar Tests', () => {
+  afterAll(() => {
+    const currentElement = document.querySelector('.progress-bar-wrapper');
+
+    if (currentElement) {
+      currentElement.parentElement.removeChild(currentElement);
+    }
+  });
+
+  describe('Progress Bar Component', () => {
+    beforeEach(function (done) {
+      setupProgressBar(templateNoAnimation).then(done).catch(err => console.error(err));
+    });
+
+    it(`Should have rendered`, () => {
+      assert(document.querySelector('progress-bar') !== null, `Progress Bar element is not rendered.`);
+    });
+  });
+
+  describe('Progress Bar Component', () => {
+    beforeEach(function (done) {
+      setupProgressBar(templateNoAnimation).then(done).catch(err => console.error(err));
+    });
+
+    it(`Should have been set to ${targetValue}% width.`, () => {
+      document.querySelector('progress-bar').setProgress(targetValue);
+      assert(document.querySelector('.progress-bar-filler').style.width === `${targetValue}%`, `Progress bar has not been set to ${targetValue}% width.`);
+    });
+  });
+
+  describe('Progress Bar Component', () => {
+    beforeEach(function (done) {
+      setupProgressBar(templateNoAnimation).then(done).catch(err => console.error(err));
+    });
+
+    it(`Should have been set to 0% width.`, () => {
+      document.querySelector('progress-bar').setProgress(-50);
+      assert(document.querySelector('.progress-bar-filler').style.width === `0%`, `Progress bar has not been set to 0% width.`);
+    });
+  });
+
+  describe('Progress Bar Component', () => {
+    beforeEach(function (done) {
+      setupProgressBar(templateNoAnimation).then(done).catch(err => console.error(err));
+    });
+
+    it(`Should have been set to 100% width.`, () => {
+      document.querySelector('progress-bar').setProgress(200);
+      assert(document.querySelector('.progress-bar-filler').style.width === `100%`, `Progress bar has not been set to 100% width.`);
+    });
+  });
+
+  describe('Progress Bar Component', () => {
+    beforeEach(function (done) {
+      setupProgressBar(templateAnimation100).then(done).catch(err => console.error(err));
+    });
+
+    it(`Should have animated to ${targetValue}% width.`, async () => {
+      const progressBar = document.querySelector('progress-bar');
+
+      progressBar.setProgress(targetValue);
+      return await createProgressBarAsyncSpec(() => {
+        assert(document.querySelector('.progress-bar-filler').style.width === `${targetValue}%`, `Progress bar has not animated to ${targetValue}% width.`);
+      });
+    });
+  });
```

Comment:
```suggestion
    it(`Should have animated to ${targetValue}% width.`, (done)  => {
      const progressBar = document.querySelector('progress-bar');

      progressBar.setProgress(targetValue);
      createProgressBarAsyncSpec(() => {
        assert(document.querySelector('.progress-bar-filler').style.width === `${targetValue}%`, `Progress bar has not animated to ${targetValue}% width.`);
        done();
      });
    });
  });
```


There were some problem with async/await, please use done.

Reply:
Also change the test bellow.

Reply:
Both changed.